### ISSUE TITLE
Failed to execute 'readAsArrayBuffer' on 'FileReaderSync' error in Windows Electron App for very large ZIM archives

### ISSUE THREAD

**Issue by Jaifroid** [kiwix/kiwix-js-windows#337]
The libzim WASM port is randomly evincing an exception when first attempting to load the full English Wikipedia ZIM in the Windows Electron app (but not the Linux Electron app), as below. Sometimes it loads correctly, and at other times it fails with this exception. **_When it fails, full-text search is unavailable._** This doesn't occur with the full Spanish Wikipedia, so it could be related to the size of the archive. The fact that it sometimes loads correctly suggests a race condition. The fact that it doesn't occur in the PWA version of the app suggests that it affects the NODEFS file system code in the WASM, but not the WORKERFS file system. However, `FileReaderSync` is a method of the standard File API that can be used in Worker context, and is not a Node-specific method.

```
Uncaught (in promise) DOMException: Failed to execute 'readAsArrayBuffer' on 'FileReaderSync':
The requested file could not be read, typically due to permission problems that have occurred after
a reference to a file was acquired.
    at Object.read (file:///C:/Users/xxxxx/AppData/Local/Programs/Kiwix%20JS%20Electron/resources/app.asar/www/js/lib/libzim-wasm.js:1:41693)
    at Object.read (file:///C:/Users/xxxxx/AppData/Local/Programs/Kiwix%20JS%20Electron/resources/app.asar/www/js/lib/libzim-wasm.js:1:57763)
    at doReadv (file:///C:/Users/xxxxx/AppData/Local/Programs/Kiwix%20JS%20Electron/resources/app.asar/www/js/lib/libzim-wasm.js:1:115113)
    at _fd_pread (file:///C:/Users/xxxxx/AppData/Local/Programs/Kiwix%20JS%20Electron/resources/app.asar/www/js/lib/libzim-wasm.js:1:115513)
    at file:///C:/Users/xxxxx/AppData/Local/Programs/Kiwix%20JS%20Electron/resources/app.asar/www/js/lib/libzim-wasm.wasm:wasm-function[600]:0x2dd27
    at file:///C:/Users/xxxxx/AppData/Local/Programs/Kiwix%20JS%20Electron/resources/app.asar/www/js/lib/libzim-wasm.wasm:wasm-function[431]:0x1fda2
    at file:///C:/Users/xxxxx/AppData/Local/Programs/Kiwix%20JS%20Electron/resources/app.asar/www/js/lib/libzim-wasm.wasm:wasm-function[1831]:0xc2149
    at file:///C:/Users/xxxxx/AppData/Local/Programs/Kiwix%20JS%20Electron/resources/app.asar/www/js/lib/libzim-wasm.wasm:wasm-function[1825]:0xc2090
    at file:///C:/Users/xxxxx/AppData/Local/Programs/Kiwix%20JS%20Electron/resources/app.asar/www/js/lib/libzim-wasm.wasm:wasm-function[475]:0x23149
    at file:///C:/Users/xxxxx/AppData/Local/Programs/Kiwix%20JS%20Electron/resources/app.asar/www/js/lib/libzim-wasm.wasm:wasm-function[1499]:0xae034
```

**Comment by Jaifroid** [comment]
Update: this appears to affect only the Windows Electron app. The Linux app appears to work fine, albeit slowly when accessing the archive over the WSL mount of the Windows NTFS File System (tbh, I'm surprised it works at all when accessing archives on the NTFS drives).

**Comment by Jaifroid** [comment]
Update: error does not seem to occur when running the Windows app unpacked, i.e. with `npm start`. This suggests that one culprit could be the `asar` internal archive, which is NOT used in the Linux app (given that AppImage is already a virtual container, so I disabled it).

**Comment by Jaifroid** [comment]
It looks like this issue is caused by the reboot that generally occurs when the Electron app is trying to reactivate the Service Worker. As we have already tried to pass a reference to the archive into the Emscripten Web Worker, the process gets repeated, causing a read error. Timing is indeed important here, and as very large archives are slower to process and access, it would explain why the error happens inconsistently but mostly with the full English Wikipedia. Hence, it's a race condition.

We either have to wait for the Service Worker to be active or else we have to fix #334, and find another way to reactivate the Service Worker in the Electron context.

Unsatisfactory workaround is to use the full English Wikipedia in jQuery mode in the Electron app for now.

**Comment by Jaifroid** [comment]
Fixed in #338.
